plugins/modules/edpm_derive_pci_passthrough_devicespec.py

```diff
@@ -0,0 +1,650 @@
+    required: true
+    default: ''
+  sriov_phydev_map:
+    description:
+      -  SR-IOV <-> phydev mapping str
+    type: str
+    required: false
+    default: ''
+  sriov_nova_conf_file:
+    description:
+      - ConfigMap file to be used in nic-partitioned deployment
+    type: str
+    required: true
+    default: '/var/lib/config-data/ansible-generated/derive_devicespec/20-sriov-nova.conf'
+'''
+
+EXAMPLES = '''
+- name: Create pci device_spec from defaults
+  edpm_nova_derive_pci_passthrough_devicespec:
+    - 04-sriov-nova.conf
+    - 'sriov1:ens2f0np0,sriov2:ens2f1np1'
+'''
+
+RETURN = '''
+rc:
+  description:
+    - Integer for the return code
+  returned: always
+  type: int
+'''
+
+_PASSTHROUGH_DEVSPEC_KEY = 'device_spec'
+_PCI_DEVICES_PATH = '/sys/bus/pci/devices'
+_SYS_CLASS_NET_PATH = '/sys/class/net'
+
+MAX_FUNC = 0x7
+MAX_DOMAIN = 0xFFFF
+MAX_BUS = 0xFF
+MAX_SLOT = 0x1F
+ANY = '*'
+REGEX_ANY = '.*'
+
+ADD_PF_PCI_ADDRESS = 0x1
+ADD_VF_PCI_ADDRESS = 0x2
+DEL_USER_CONFIG = 0x3
+KEEP_USER_CONFIG = 0x4
+
+
+class InvalidConfigException(ValueError):
+    pass
+
+
+def get_pci_field_val(
+        prop: str, maxval: int, hex_value: str):
+    if prop == ANY:
+        return REGEX_ANY
+    try:
+        v = int(prop, 16)
+    except ValueError:
+        raise InvalidConfigException('Invalid PCI address specififed {!r}'.format(prop))
+    if v > maxval:
+        raise InvalidConfigException('PCI address specififed {!r} is out of range'.format(prop))
+    return hex_value % v
+
+
+def get_pciaddr_dict_from_usraddr(pci_addr: str):
+    """Convert PCI address in STRING format to DICT
+    (this is done for uniformity in PCI address)
+    """
+    pci_dict = {}
+    dbs, sep, func = pci_addr.partition('.')
+    pci_dict['function'] = ANY
+    if func:
+        func = func.strip()
+        pci_dict['function'] = func
+    if dbs:
+        dbs_fields = dbs.split(':')
+        if len(dbs_fields) > 3:
+            raise InvalidConfigException('Invalid PCI address specififed {!r}'.format(pci_addr))
+        # If we got a partial address like ":00.", we need to turn this
+        # into a domain of ANY, a bus of ANY, and a slot of 00. This code
+        # allows the address,bus and/or domain to be left off
+        dbs_all = [ANY] * (3 - len(dbs_fields))
+        dbs_all.extend(dbs_fields)
+        dbs_checked = [s.strip() or ANY for s in dbs_all]
+
+        ''' domain, bus, slot = dbs_checked '''
+        pci_dict['domain'], pci_dict['bus'], pci_dict['slot'] = dbs_checked
+
+    pci_dict['domain'] = get_pci_field_val(pci_dict['domain'], MAX_DOMAIN, '%04x')
+    pci_dict['slot'] = get_pci_field_val(pci_dict['slot'], MAX_SLOT, '%02x')
+    pci_dict['bus'] = get_pci_field_val(pci_dict['bus'], MAX_BUS, '%02x')
+    pci_dict['function'] = get_pci_field_val(pci_dict['function'], MAX_FUNC, '%1x')
+    return pci_dict
+
+
+def get_pci_regex_pattern(config_regex: str, size: int, maxval: int, hex_value: str):
+    if config_regex in [ANY, REGEX_ANY]:
```

Comment:
@vcandapp same description and also either we can specify type for argument variables in all the functions or remove it. looks like not consistent.